### PR TITLE
Salingmw code review 2

### PR DESCRIPTION
**Pull request by salingmw**
Analysis: This program is an app that will take the chemicals that the user inputs and will recommend what combination of other chemicals will be needed to clean the pool and make it safer to use.

The github was available early and on time.

The codebase was extremely well commented and almost all of the classes and functions were commented on. Each comment was well written and describes exactly what is going on. The program compiled.

The first change I made was changing the File and class name to proper Upper Camel Case to follow the coding conventions and style guidelines for Kotlin.

The second change I made was to add ASINTag and ASINTiers to the ExampleUnitTest file to make the file correctly compile and fix the Unit Tests.

The third change I made was updating and changing the names of the individual Algae's  to make the more descriptive and make them easier to understand. When I read the bAlgae I thought it was blue but after looking further it turns out that it was black.

THINGS I LEARNED:
While looking through the code I learned some more things about the coding guidelines and conventions.
I learned the proper syntax for using arrays
I learned that read-only properties can be used instead of functions in some cases according to the Kotlin Coding Conventions

My Commits
1) https://github.com/erswagner7/PuzzlePic/commit/de0667582a96aed37c40fd1914492f608202b057
2) https://github.com/erswagner7/PuzzlePic/commit/8221aa4bd2d5c5d849612182ce1e05e71422feca
3) https://github.com/erswagner7/PuzzlePic/commit/d0984af69359dd1c91ba0b04db7202ce6467a28f

### CODE REVIEW

File: app/src/main/java/com/example/pool/MainActivity.kt [hunk 1]
```diff
@@ -33,9 +33,9 @@ class MainActivity : AppCompatActivity() {
     val phosphates = Chemical(name= "pho", okRange= arrayOf(0F, 100F), hoursCantSwim= 0F,
             ozPerGallon= .005F, ASINTiers= arrayOf("N/A"))
 
-    val gAlgae = Algae(type= "Green", hoursCantSwim= 0F, ozPerGallon= 0F, chlBoostPerGallon= 0F, ASINTag = "B002WKJAYS")
-    val yAlgae = Algae(type= "Yellow", hoursCantSwim= 0F, ozPerGallon= 0F, chlBoostPerGallon= 0F, ASINTag = "B01LW1QNZ7")
-    val bAlgae = Algae(type= "Black", hoursCantSwim= 0F, ozPerGallon= 0F, chlBoostPerGallon= 0F, ASINTag = "B00BGNLPCW")
+    val greenAlgae = Algae(type= "Green", hoursCantSwim= 0F, ozPerGallon= 0F, chlBoostPerGallon= 0F, ASINTag = "B002WKJAYS")
+    val yellowAlgae = Algae(type= "Yellow", hoursCantSwim= 0F, ozPerGallon= 0F, chlBoostPerGallon= 0F, ASINTag = "B01LW1QNZ7")
+    val blackAlgae = Algae(type= "Black", hoursCantSwim= 0F, ozPerGallon= 0F, chlBoostPerGallon= 0F, ASINTag = "B00BGNLPCW")
```
Comment (salingmw) [inline app/src/main/java/com/example/pool/MainActivity.kt:38]: Updated the names of the Algaes to make them more descriptive and easier to understand per coding conventions https://kotlinlang.org/docs/coding-conventions.html#functions-vs-properties

File: app/src/main/java/com/example/pool/PoolStatusItem.kt [hunk 2]
```diff
@@ -3,5 +3,5 @@ package com.example.pool
 import android.text.Editable
 import org.w3c.dom.Text
 
-data class poolStatusItem(val imageResource: Int, val Title: String, val description: String, var poolStatus: String) {
+data class PoolStatusItem(val imageResource: Int, val Title: String, val description: String, var poolStatus: String) {
```
Comment (salingmw) [inline app/src/main/java/com/example/pool/PoolStatusItem.kt:6]: Updated the name of PoolStatusItem to make it follow the Upper Camel Case via the coding conventions https://kotlinlang.org/docs/coding-conventions.html#functions-vs-properties

File: app/src/test/java/com/example/pool/ExampleUnitTest.kt [hunk 3]
```diff
@@ -19,13 +19,13 @@ class ExampleUnitTest {
...
     fun chemicalDTO_toStringFormat() {
-        val chemical = Chemical(name="example", okRange=arrayOf(0F, 1F), hoursCantSwim=1F, ozPerGallon=.5F)
+        val chemical = Chemical(name="example", okRange=arrayOf(0F, 1F), hoursCantSwim=1F, ozPerGallon=.5F, ASINTiers=arrayOf("B084GPWRBL", "B08PG4C2NQ", "B004WDVT6K","B084GPS6KR", "B077715Y9L", "B07YZPNWDL"))
         assertTrue(chemical.toString().equals("example is a chemical with an acceptable range of 0 to 1" +
                 "and is unsafe for pool goers 1 hour after use."))
     }
     @Test
     fun algaeDTO_toStringFormat() {
-        val algae = Algae(type="example", hoursCantSwim=0F, ozPerGallon=.5F, chlBoostPerGallon=1F)
+        val algae = Algae(type="example", hoursCantSwim=0F, ozPerGallon=.5F, chlBoostPerGallon=1F, ASINTag="B002WKJAYS")
```
Comment (salingmw) [inline app/src/test/java/com/example/pool/ExampleUnitTest.kt:28]: Added the ASINTag and ASINTiers to the Unit Tests to solve the errors that came with this file. I followed the coding conventions via https://kotlinlang.org/docs/coding-conventions.html#functions-vs-properties

File: app/src/main/java/com/example/pool/MainActivity.kt [hunk 4]
```diff
@@ -81,13 +81,13 @@ class MainActivity : AppCompatActivity() {
...
-            return gAlgae.ASINTag
+            return greenAlgae.ASINTag
         //1 qt
         else if (productType=="yellowAlgaecide")
-            return yAlgae.ASINTag
+            return yellowAlgae.ASINTag
         //1 qt
         else if (productType=="blackAlgaecide")
-            return bAlgae.ASINTag
+            return blackAlgae.ASINTag
```
Comment (salingmw) [inline app/src/main/java/com/example/pool/MainActivity.kt:90]: Updated the names of the algaes in the affected files from the changes above.

File: app/src/main/java/com/example/pool/PoolItemAdapter.kt [hunk 5]
```diff
@@ -12,7 +12,7 @@ import kotlinx.android.synthetic.main.pool_status_item.view.*
 
 
 
-class PoolItemAdapter(private val poolList: List<poolStatusItem>): RecyclerView.Adapter<PoolItemAdapter.PoolItemViewHolder>() {
+class PoolItemAdapter(private val poolList: List<PoolStatusItem>): RecyclerView.Adapter<PoolItemAdapter.PoolItemViewHolder>() {
```
Comment (salingmw) [inline app/src/main/java/com/example/pool/PoolItemAdapter.kt:15]: Updated the name in the affected file from the change above

File: app/src/main/java/com/example/pool/MainActivity.kt (hunk 1 above)
Comment (discospiff) [reply app/src/main/java/com/example/pool/MainActivity.kt:38]: Good idea.  g could be green or gray; b could be blue or black.  Spell them out, so there is no ambiguity.

File: app/src/main/java/com/example/pool/PoolStatusItem.kt (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/com/example/pool/PoolStatusItem.kt:6]: Yes... UpperCamelCase is correct for classes.

**Comment by discospiff** [comment]
Several good changes in this branch, which reduce technical debt.  The team should consider merging.

**Pull request merged**